rust/experimental/server/src/store/memory.rs

```diff
@@ -413,6 +418,14 @@ impl Store for MemoryStore {
                                 last_block_id = -1;
                             }
                         }
+                        candidate_blocks = candidate_blocks
+                            .into_iter()
+                            .filter(|block| {
+                                block_ids_filter
+                                    .as_ref()
+                                    .map_or(true, |filter| filter.contains(block.block_id as u64))
+                            })
+                            .collect();
                     }
                 }
 
```

Comment:
How about putting this filter here? 

```
 candidate_blocks = candidate_blocks
                            .into_iter()
                            .filter(|block| {
                                block_ids_filter
                                    .as_ref()
                                    .map_or(true, |filter| filter.contains(block.block_id as u64))
                            })
                            .collect();
```

Reply:
Otherwise, when the last_block_id = -1, the filter is not valid. This case also should be added into test.

Reply:
sure